### DISCUSSION TITLE
Got os error 10054

### DISCUSSION THREAD

**Discussion by wubushanyan** [rustdesk/rustdesk#10082] [Q&A]
## Environment
Client Version:1.3.3

Server Version : Debain12 with latest docker

Windows11 -> Windows10 LTSC 21H2

## What I see 

I found two errors during the connection process

I have 2 pc, once in home once in my workspace

Set home to PC-A(Win10)   another is PC-B(Win11)

PC-A connect to PC-B I got below 
![image](https://github.com/user-attachments/assets/01d5f949-9bca-49c4-a3ec-296e4348ff2b)

PC-B connect to PC-A I got below
![image](https://github.com/user-attachments/assets/d7b78e15-a7ef-4693-a2ea-1d659599ba0d)

## What I do
Deploy  rustdesk server with docker in debain12 VPS. Confirm it can be connected by client.

Setting 2 PC client with server ip and key

Input ID to connect then I got error.

**Already read FAQ, set `/r` behind ID to force relay, but not work.**
![image](https://github.com/user-attachments/assets/18653aa3-a30f-4e4d-9c37-bf4e746f3f45)

**Comment by rustdesk** [comment]
How about 1.3.2 on controlled side?

**Reply by wubushanyan** [reply]
had try, same as 1.3.3, I update client jsut 2hours ago

**Comment by rustdesk** [comment]
Please paste your access log of 1.3.3 on controlled side. https://github.com/rustdesk/rustdesk/wiki/FAQ#access-logs

**Reply by 21pages** [reply]
Could you find some clue from the Windows task manager every time you connect? You can see the command line of the program if you right click the column header and select "Command line".


Make a screenshot like this.
![image](https://github.com/user-attachments/assets/7edde642-f680-43b3-a99a-f1975ff4d8e8)

**Reply by wubushanyan** [reply]
what should I pay attention to? It seems normal
![image](https://github.com/user-attachments/assets/c3d7de24-6224-426e-a48f-d237a42eb7b8)

**Reply by wubushanyan** [reply]
![image](https://github.com/user-attachments/assets/b5529d63-de36-41c6-8e0c-a5ea61ef5596)

**Reply by 21pages** [reply]
1. Also select the "PID" column.
2. Let the screenshot pc to be the controlled side, kill the "RustDesk.exe" without any command line, then connect to it from another pc.
3. I want to confirm if there is a zombie process, or if the "--server" process is crashed in a loop, or the PID of the "--server" process will change after the access failure.

**Reply by wubushanyan** [reply]
Front screenshot was from the control side

Behind screenshot from the controlled, sry for I provided the wrong environment info
my PC-A is win10 LTSC 21H2
![image](https://github.com/user-attachments/assets/ebd87f8f-f709-4631-ab86-c5f041d0b443)


1. Also select the "PID" column.
    **Done**
2. Let the screenshot pc to be the controlled side, kill the "RustDesk.exe" without any command line, then connect to it from another pc.
    **Done  not to kill and try connect to PC-A(controlled)**  
3. I want to confirm if there is a zombie process, or if the "--server" process is crashed in a loop, or the PID of the "--server" process will change after the access failure.
    **PID not change**

**Reply by 21pages** [reply]
>  [src\server.rs:492] ipc is occupied by another process, try kill it

On the controlled side, kill "--server" process manually, a new log file `RustDesk_rCurrent.log` will be generated, the old log file is archived, does this line still appear in the latest log of `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\server\RustDesk_rCurrent.log` ?

**Reply by 21pages** [reply]
Can you check if the "hbbr" process still exists on your debian server? If exists, check whether port 21117 is open.

**Reply by wubushanyan** [reply]
looks fine

![image](https://github.com/user-attachments/assets/c5578b2e-fcec-43a4-8315-1937fa662950)


![image](https://github.com/user-attachments/assets/668bbbb0-66e7-4a5c-bee3-62cab0641336)

**Reply by 21pages** [reply]
Is the network condition stable? Asking because there are many relay request logs in short time.

**Reply by wubushanyan** [reply]
> Is the network condition stable? Asking because there are many relay request logs.

yes 
tryed sunlogin it was stable

**Reply by 21pages** [reply]
The network condition between debian and your computers. Could you try another time to check if the error message is still `Failed to receive public key`?

**Reply by 21pages** [reply]
Could you show the docker logs?

**Reply by 21pages** [reply]
Did `ipc is occupied by another process, try to kill it` appear many times in today's logs?

**Reply by wubushanyan** [reply]
nop  
![image](https://github.com/user-attachments/assets/662f0488-1e88-4d95-a106-53007c395e34)

**Reply by 21pages** [reply]
In `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\server\`

**Reply by 21pages** [reply]
Did you do something at 21:46:40? "--server" is restarted at that time, but the ipc is occupied by main window process.

**Reply by wubushanyan** [reply]
Maybe my worksapce PC shutdown I had home now

**Reply by 21pages** [reply]
I want to find the reason why the ipc of `rustdesk --server` is often occupied by rustdesk main window, could you help me? 
I need to know what happens at that moment. It may happen when the main window process is started before the --server process, or there is a main window zombie process, which occurs when the system is started or rustdesk installation.
Here are two powershell script, 
1. `print_ipc.ps1`: It will print logs to `ipc_occupied_lines.txt`, paste the content, it is the history.
2. `monitor_ipc.ps1`: if `ipc is occupied by another process, try kill it` occurs on the lastest log file, it will popup a powershell messagebox, then we know what happens at that moment. It has a system sound, no need to pay atttention to it, just run it when system start.

[print_ipc.zip](https://github.com/user-attachments/files/17953237/print_ipc.zip)
[monitor_ipc.zip](https://github.com/user-attachments/files/17953381/monitor_ipc.zip)

![image](https://github.com/user-attachments/assets/eda0a190-14a8-48bf-bf6b-442c3981d0d1)

**Reply by wubushanyan** [reply]
Had run 2 ps script, but not see powershell messagebox popup. Only got a txt below.
![image](https://github.com/user-attachments/assets/7f99ea63-37ac-4b91-bc4a-a57a4bf83af5)

[ipc_occupied_lines.txt](https://github.com/user-attachments/files/17954360/ipc_occupied_lines.txt)

**Reply by 21pages** [reply]
Thanks,
1. Can you recall what occurred during those moments in the file? It occurs often on your computer.
2. Please keep `monitor_ipc.ps1` running in the background; a message box will pop up when an IPC occupy occurs.

**Reply by wubushanyan** [reply]
noting special  I restart my pc to catch new log

but I can confirm the  monitor_ipc.ps1 not popup new messagebox while ipc occupy
new log shows in 2024-11-29 13:56:52.179082 +08:00] I got ipc occupy, but nothing popup

I just open rustdesk client, try to connect and high speed click the reconnect button  XD
![image](https://github.com/user-attachments/assets/ecba4170-02be-4aed-a605-8553c1c308ed)


[ipc_occupied_lines.txt](https://github.com/user-attachments/files/17955034/ipc_occupied_lines.txt)

**Reply by 21pages** [reply]
> noting special I restart my pc to catch new log

Sorry for that, I test it via changing the content of  `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\server\RustDesk_rCURRENT.log` manually.

>  restart my pc

IPC occupy happens on system restart or failed connection? Your system was restarted many times yesterday?

> high speed click the reconnect button

I'll try to fix the big number.

**Reply by wubushanyan** [reply]
yesterday only once

**Reply by 21pages** [reply]
Could you upload a zip of folder `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\service`? There is no personal informations in that folder.

**Reply by wubushanyan** [reply]
here u are
[service.zip](https://github.com/user-attachments/files/17955357/service.zip)

**Reply by 21pages** [reply]
` Got service control event: Stop` is printed many times. Did you uninstall rustdesk or stop rustdesk service many times yesterday?

**Reply by wubushanyan** [reply]
may the update progress do it? 
Yesterday I do install rustdesk 1.3.2 the at 5p.m may be. I found it can update to 1.3.3.
Then I just update


Not uninstall， but some time I open rustdesk the service not up, I should manually up it by click a tips which at the bottom of the window of rustdesk client

**Reply by 21pages** [reply]
>  but some time I open rustdesk the service not up

Does this happen many times? It's unreasonable, I'll give you another powershell to monitor the three rustdesk processes, hope it can run on your computer.

**Reply by wubushanyan** [reply]
no I only update once

**Reply by 21pages** [reply]
Okay, here is the PowerShell command. RustDesk needs to be installed in the default path: `C:\Program Files\RustDesk`. Additionally, please ensure that the shell is run as an administrator. The log file is `monitor_rustdesk_process.log`.

[monitor_rustdesk_process.zip](https://github.com/user-attachments/files/17956264/monitor_rustdesk_process.zip)

**Reply by wubushanyan** [reply]
Might upload new log next week, I got weekend off

**Reply by 21pages** [reply]
Have a good weekend!
